std/range/package.d

```diff
@@ -6629,10 +6627,20 @@ if (isForwardRange!RangeOfRanges &&
         }
     }
 
-    // ElementType opIndex(size_t n)
-    // {
-    //     return _input[n].front;
-    // }
+    static if (isRandomAccessRange!RangeOfRanges)
+    {
+        auto ref opIndex(size_t n)
+        {
+            auto r = this.save;
+            r._input = _input.dup;
```

Comment:
This isn't necessary, nor is it supported by forward ranges in general.

Reply:
If you eliminate that line and change the result of this.save it will affect the original instance as well, due to how `@property Transposed save()` is currently implemented.

Reply:
Not sure what you mean? if save doesn't make it so you can iterate independently of the original, then it's not implemented correctly. But you definitely should be able to change the underlying values via `opIndex`.

For example, this should work:
```D
int[][] x = [[1,2,3],[4,5,6]];
auto t = x.transposed;
x[0][1] = 10;
assert(t == [[1,4],[10,5],[3,6]]);
t[2][1] = 15;
assert(x == [[1,10,3],[4,5,15]]);
```

Reply:
What I meant, without that line we have the following behaviour:
`
auto x = [[1,2,3],[4,5,6]].transposed;
writeln(x[1]); // prints [2,5] OK
writeln(x[1]); // prints [3,6] NOT OK
`

Which I assume is not ok?

Reply:
Oh, definitely that's not OK. Let me think about this a bit, as the root cause isn't apparent to me. I don't think this is just a problem with `opIndex`.

Reply:
Yeah, this is basically a problem of `save`:

```D
auto x = [[1,2,3],[4,5,6]];
auto y = x.save;
y.popFront;
writeln(x.front); // prints [2,5]
```
Clearly, save doesn't work here the way it's implemented, this is a bug in the original code.

Reply:
https://issues.dlang.org/show_bug.cgi?id=17952